ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -11677,11 +11677,19 @@ void HqlCppTranslator::buildCppFunctionDefinition(BuildCtx &funcctx, IHqlExpress
     }
 
     funcctx.addQuotedCompound(proto);
+    funcctx.addQuoted("#if defined(__clang__) || (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))\n"
```

Comment:
According to this https://gcc.gnu.org/ml/gcc-help/2011-01/msg00112.html push wasn't supported until 4.6